sdk/src/main/java/com/horizen/account/event/EthereumEvent.java

```diff
@@ -0,0 +1,133 @@
+            if (acObj instanceof Field) {
+                var field = (Field) acObj;
+                annotatedParams.put(acObj.getAnnotation(Parameter.class).value(), new EventParameterData(indexed, field.getType(), (Type) field.get(eventInstance)));
+            } else {
+                var method = (Method) acObj;
+                annotatedParams.put(acObj.getAnnotation(Parameter.class).value(), new EventParameterData(indexed, method.getReturnType(), (Type) method.invoke(eventInstance)));
+            }
+        }
+        return annotatedParams;
+    }
+
+    /**
+     * @param contractAddress
+     * @param eventFunction
+     * @param anonymous
+     * @return EvmLog containing the contract address, the topics and the data
+     * @throws IOException
+     */
+    private static EvmLog createEvmLog(Address contractAddress, Function eventFunction, Boolean anonymous) throws IOException {
+        var address = com.horizen.evm.utils.Address.FromBytes(Numeric.hexStringToByteArray(contractAddress.getValue()));
+        List<Hash> topics = new ArrayList<>();
+        ByteArrayOutputStream dataOutputStream = new ByteArrayOutputStream();
+        var outputParameters = eventFunction.getOutputParameters();
+
+        if (!anonymous) {
+            topics.add(Hash.FromBytes(Numeric.hexStringToByteArray(EventEncoder.encode(new Event(eventFunction.getName(), new ArrayList<>(outputParameters))))));
+        }
+
+        for (var i = 0; i < outputParameters.size(); i++) {
+            var encodedValue = Numeric.hexStringToByteArray(TypeEncoder.encode(eventFunction.getInputParameters().get(i)));
+            if (outputParameters.get(i).isIndexed()) {
+                // values <= 32 byte will be used as is
+                if (encodedValue.length > 32) encodedValue = (byte[]) Keccak256.hash(encodedValue);
+                topics.add(Hash.FromBytes(encodedValue));
+            } else {
+                dataOutputStream.write(encodedValue);
+            }
+        }
+        if (topics.size() > 4)
+            throw new IllegalArgumentException("Error: More than four topics defined - defined topics: " + topics.size());
+        return new EvmLog(address, topics.toArray(new Hash[topics.size()]), dataOutputStream.toByteArray());
+    }
+
+    /**
+     * @param contractAddress
+     * @param eventInstance   Instance of the custom event class
+     * @return EvmLog containing the contract address, the topics and the data
+     * @throws ClassNotFoundException
+     * @throws IOException
+     * @throws IllegalAccessException
+     * @throws InvocationTargetException
+     */
+    public static EvmLog getEvmLog(Address contractAddress, Object eventInstance) throws ClassNotFoundException, IOException, IllegalAccessException, InvocationTargetException {
+        List<TypeReference<?>> parametersTypeRef = new ArrayList<>();
+        List<Type> convertedParams = new ArrayList<>();
+        var annotatedParameters = getEventParameterData(eventInstance);
```

Comment:
Not clear to me why `getEventParameterData()` returns a map, we use `annotatedParameters` taking the values only